tests/conftest.py

```diff
@@ -27,6 +24,20 @@ def tokens(create_token, payer, request):
     return [create_token(payer) for _ in range(int(request.param.split(" ")[0]) + 1)]
 
 
+@pytest.fixture(scope="session")
+def token(tokens):
+    if len(tokens) == 0:
+        pytest.skip("No valid tokens")
+
+    return tokens[0]
```

Comment:
Perf on the test would be to use try/ except because i think token lists len can be slow sometimes

On except, do the pytest.skip